nixos/modules/services/web-apps/funkwhale.nix

```diff
@@ -0,0 +1,378 @@
+      };
+    in
+    {
+      funkwhale-setup = {
+        description = "Funkwhale setup and migrations";
+        before = [
+          "funkwhale-beat.service"
+          "funkwhale-server.service"
+          "funkwhale-worker.service"
+        ];
+        script = ''
+          if [ -z ''${DJANGO_SECRET_KEY+x} ]; then
+            (
+              umask 0077
+              ${cfg.package.python.interpreter} -c "from django.core.management.utils import get_random_secret_key; print(f'DJANGO_SECRET_KEY=\"{get_random_secret_key()}\"')" > $STATE_DIRECTORY/secret_key
+            )
+            export DJANGO_SECRET_KEY="$(cat $STATE_DIRECTORY/secret_key)"
+          fi
+
+          mkdir -p $STATE_DIRECTORY/bin
+          ln -sf ${manage} $STATE_DIRECTORY/bin/funkwhale
+
+          ${cfg.package}/bin/funkwhale migrate
+          ${cfg.package}/bin/funkwhale collectstatic --no-input
+
+          export DJANGO_SUPERUSER_USERNAME="${cfg.superUserName}"
+          export DJANGO_SUPERUSER_PASSWORD="$(cat ${cfg.superUserPasswordFile})"
+          export DJANGO_SUPERUSER_EMAIL="${cfg.superUserEmail}"
+
+          if [ ! -e "$STATE_DIRECTORY/.${cfg.superUserName}_created" ]; then
+            ${cfg.package}/bin/funkwhale createsuperuser --noinput
+            touch "$STATE_DIRECTORY/.${cfg.superUserName}_created"
+          fi
+        '';
+        serviceConfig = {
+          Type = "oneshot";
+        } // commonServiceConfig;
+      } // commonUnitConfig;
+
+      funkwhale-server = {
+        description = "Funkwhale application server";
+
+        serviceConfig = {
+          ExecStart = ''
+            ${cfg.package.python.pkgs.gunicorn}/bin/gunicorn config.asgi:application \
+              --name=funkwhale-server \
+              --worker-class=uvicorn.workers.UvicornWorker \
+              --workers=${cfg.settings.FUNKWHALE_WEB_WORKERS} \
+              --bind=${cfg.settings.FUNKWHALE_API_IP}:${cfg.settings.FUNKWHALE_API_PORT} \
+              --pythonpath="${cfg.package.out}/lib/funkwhale/api"
+            '';
+        } // commonServiceConfig;
+      } // commonUnitConfig;
+
+      funkwhale-beat = {
+        description = "Funkwhale celery beat process";
+
+        serviceConfig = {
+          ExecStart = ''
+            ${cfg.package.python.pkgs.celery}/bin/celery \
+              --app funkwhale_api.taskapp \
+              beat \
+              --loglevel INFO
+          '';
+        } // commonServiceConfig;
+      } // commonUnitConfig;
+
+      funkwhale-worker = {
+        description = "Funkwhale celery worker";
+        serviceConfig = {
+          ExecStart = ''
+            ${cfg.package.python.pkgs.celery}/bin/celery \
+              --app funkwhale_api.taskapp \
+              worker \
+              --concurrency=${cfg.settings.CELERYD_CONCURRENCY} \
+              --loglevel INFO
+          '';
+        } // commonServiceConfig;
+      } // commonUnitConfig;
+    };
+
+
+    services.nginx = {
+      enable = mkDefault true;
```

Comment:
We should add an option wether nginx should be enabled which also disables the other options if people want to do with another reverse proxy server.